### PR TITLE
Rewrite @Unroll tests

### PR DESCRIPTION
**Pull request by blindpirate**
This is to fix 'Received a failure event for test with unknown id '2.9'' error
which is cuased by non-unique Spock @Unroll test names.

Fixes #?

See https://builds.gradle.org/viewLog.html?buildId=35948514&buildTypeId=Gradle_Check_VfsRetention_30_bucket48&tab=buildResultsDiv

and https://github.com/gradle/gradle/issues/8787

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test QF

File: subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy [hunk 1]
```diff
@@ -50,6 +49,7 @@ class BuildActionSerializerTest extends SerializerSpec {
         result.startParameter.taskNames == ['a', 'b']
     }
 
+    @Unroll
```
Comment (wolfs) [inline subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy:52]: Why did you remove the `@Unroll` from the class?

Comment (jjohannes) [reply subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy:52]: I assume because the other tests don't need to unroll. It seems to be the pattern we usually use (annotate only the methods that actually `Unroll`). Or should we do it differently? (I don't care much, I just like consistency :)).

**Comment by ldaley** [comment]
> I assume because the other tests don't need to unroll. It seems to be the pattern we usually use (annotate only the methods that actually Unroll).

My experience with this is that the net result is lots of tests with missing `@Unroll`, which makes the diagnostics worse.

File: subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy (hunk 1 above)
Comment (ldaley) [reply subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy:52]: I assume because the other tests don't need to unroll. It seems to be the pattern we usually use (annotate only the methods that actually Unroll).

My experience with this is that the net result is lots of tests with missing @Unroll, which makes the diagnostics worse.

Comment (wolfs) [reply subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy:52]: +1 to what @ldaley said: When we don't put the `@Unroll` on the class most times we end up missing with an `@Unroll` on a test method. That is why I started moving the `@Unroll` on the class level when we change a test class. 

Is there any disadvantage for the `@Unroll` on the class level? There was a problem with IDEA navigation from the test to the source code, though that has been fixed for a very long time now.

Comment (blindpirate) [reply subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/BuildActionSerializerTest.groovy:52]: No disadvantange for `@Unroll` on test class level, just because only one or two test methods in this class have `where` block. We can put the `@Unroll` back of course.

**Comment by blindpirate** [comment]
Thanks @jjohannes @ldaley @wolfs !
